src/components/select/SelectItemCommons.tsx

```diff
@@ -42,13 +42,30 @@ function withSelectItemWrapper<P extends SelectOption>(
   optionType: SelectOptionType,
 ) {
   const Wrapper = forwardRef<HTMLDivElement, P>((props, ref) => {
-    const { helpText, label, prefix, suffix, value, ...rest } = props;
+    const { helpText, label, prefix, selected, suffix, value, ...mantineAndCustomProps } = props;
+
+    const {
+      'aria-selected': _ariaSelected,
+      className,
+      'data-disabled': _dataDisabled,
+      'data-hovered': _dataHovered,
+      'data-selected': _dataSelected,
+      onMouseDown,
+      onMouseEnter,
+      tabIndex,
+      ...customProps
+    } = mantineAndCustomProps;
+
     return (
-      <SelectItemWrapper ref={ref} {...rest}>
-        <SelectItemStatus optionType={optionType} {...props} />
+      <SelectItemWrapper ref={ref} selected={selected} {...mantineAndCustomProps}>
+        <SelectItemStatus optionType={optionType} selected={selected} />
         {prefix}
         <SelectItemInner>
-          {WrappedComponent ? <WrappedComponent {...props} /> : <span>{props.label}</span>}
+          {WrappedComponent ? (
+            <WrappedComponent {...({ label, selected, value, ...customProps } as P)} />
```

Comment:
We can only pass `...customProps` here if we don't extract `label`, `selected` and `value`. Also, maybe the `as P` will become useless.